src/middlewared/middlewared/plugins/virt/device.py

```diff
@@ -14,6 +14,8 @@
 
 class VirtDeviceService(Service):
 
+    PCI_CHOICES = None
```

Comment:
I don't like class attributes for caching values. Please use the `@cache` decorator. We have a custom one in `middlewared.utils.functools` module. Also, is there any way we can return an immutable object here so that this cache doesn't get altered? I'm thinking of using something like `dataclasses(frozen=True, slots=True)`.

Something like this
```
from dataclasses import dataclass, asdict

@dataclass(slots=True, frozen=True, kw_only=True)
class PciEntry:
    ...


@cache
def get_pci_devs_cache(self) -> tuple[PciEntry]:
     result = list()
     for blah, blah in blah.items():
         result.append(PciEntry(blah, blah, blah))
     return tuple(result)
  
@api_method(...)
 def pci_choices():
     return [asdict(i) for i in self.get_pci_devs_cache()]